### DISCUSSION TITLE
Will be awesome to profile container process directly

### DISCUSSION THREAD

**Discussion by tufeigunchu** [google/orbit#4638] [Ideas]
When I run OrbitService on the host and profile a container task, it can't resolve symbols.

**Comment by florian-kuebler** [comment]
Thanks for reaching out. https://github.com/google/orbit/discussions/4532 is going into the same direction.

But, basic profiling (sampling, scheduling info, kernel-based dynamic insturmentation) should work from the host. I am a bit puzzled why you are not getting symbols. Can you post a logfile (stdout from Orbit)?

**Comment by pierricgimmig** [comment]
Profiling containers will not work out of the box. The module paths will be expressed in the container's filesystem, Orbit will try to treat them as host paths. This will not work. There is a way to make it work, but it involves some non-trivial changes. Injecting libraries into the container is also problematic because the OS of the container might not match that of the host and there might be dependency mismatches for the injected libraries.

**Comment by florian-kuebler** [comment]
>The module paths will be expressed in the container's filesystem, Orbit will try to treat them as host paths. This will not work.

Right. However, if you do have the symbols files available at the environment where you run the client/ui part of Orbit, you can simply specify the symbol location as shown in our [documentation](https://github.com/google/orbit/blob/main/documentation/DOCUMENTATION.md#additional-symbol-locations).

As @pierricgimmig stated, any feature that requires injection will have problems. Namely, these are manual instrumentation, user-space dynamic instrumentation (the "kernel" option should still work), as well as the Vulkan layer.
